File: framework/languages/java/zlink-framework-core/src/main/java/systems/zlink/framework/actors/ZLinkActorContext.java
package systems.zlink.framework.actors;

import java.util.Optional;

public interface ZLinkActorContext {
    String actorId();

    long objectGeneration();

    String meshName();

    Optional<String> spotId();

    ZLinkBoundSession boundSession();

    ZLinkActorJoinCall joinSpot(String spotId);

    ZLinkActorJoinCall joinSpot(String spotId, Object request);

    ZLinkActorJoinCall joinEntrySpot();

    ZLinkActorJoinCall joinEntrySpot(Object request);
}

File: framework/languages/node/packages/framework/src/contracts/Actors/ZLinkActorContext.ts
import type { SpotId } from '../Common';
import type { ZLinkBoundSession } from '../Streams';
import type { ZLinkActorJoinEntrySpotCall, ZLinkActorJoinSpotCall } from './ZLinkActorFactory';

export interface ZLinkActorContext {
  readonly actorId: string;
  readonly objectGeneration: bigint;
  readonly meshName: string;
  readonly spotId?: SpotId;
  readonly boundSession: ZLinkBoundSession;
  joinSpot(spotId: SpotId): ZLinkActorJoinSpotCall;
  joinSpot(spotId: SpotId, request: unknown): ZLinkActorJoinSpotCall;
  joinEntrySpot(): ZLinkActorJoinEntrySpotCall;
  joinEntrySpot(request: unknown): ZLinkActorJoinEntrySpotCall;
}
